src/components/atoms/Selection/Selection.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <div class="w-[210px]">
+    <label class="text-left text-black w-[100%] truncate" v-if="props.selection.label">
+      {{ props.selection.label }}
+      <span v-if="props.selection.asterisk" class="text-red-500">*</span>
+    </label>
+
+    <div :class="['cursor-pointer', { 'pointer-events-none opacity-50': props.selection.isDisabled }]">
+      <!-- selected -->
+      <div
+        class="flex justify-between items-center w-[100%] h-[42px] px-[10px] border-[1px] border-gray-300 rounded-md"
+        @click="handleIsOpen"
+      >
+        <span v-if="selectedValue?.value">{{ selectedValue.label || props.selection?.valueList[0]?.label }}</span>
+        <div class="duration-300 " :class="{ '-rotate-180': isOpen }">
+          <i class="fa-solid fa-chevron-down mr-2 text-gray-300 "></i>
+        </div>
+      </div>
+      <!-- end select -->
+
+      <!-- error -->
+      <p class="text-[10px] text-red-500" v-if="props.selection?.error">{{ props.selection?.helperText }}</p>
+      <!-- end error -->
+
+      <!-- option -->
+       <div :class="[{'animate-scaleShowSelect':isOpen} , {'animate-unScaleShowSelect':!isOpen} ]">
+        <div v-if="isOpen" class="overflow-hidden w-[100%] mt-[5px] border-[1px] border-gray-200 rounded-md">
+          <div
+            v-for="option in props.selection.valueList"
+            :key="option.label"
+            @click="handleSetValueSelected(option)"
+            :class="[
+              { 'font-bold bg-gray-100': option.value === selectedValue.value },
+              'w-[100%]',
+              'h-[42px]',
+              'px-[10px]',
+              'flex',
+              'justify-start',
+              'items-center',
+              'hover:bg-gray-100'
+            ]"
+          >
+            {{ option.label }}
+          </div>
+        </div>
+       </div>
+      <!-- end option -->
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { Ref, ref } from 'vue'
+
+interface SelectionProp {
+  selection: {
+    label?: string
+    asterisk?: boolean
+    placeholder?: string
+    valueList: {
+      label: string
+      value: string
+    }[]
+    value: { label: string; value: string }
+    error?: boolean
+    helperText?: string
+    ref?: Ref<any>
+    isDisabled?: boolean
+  }
+}
+const props = withDefaults(defineProps<SelectionProp>(), {
+  selection: {
+    valueList: [],
+    value: { label: '', value: '' }
+  }
+})
+
+const selectedValue = ref( props.selection?.valueList[0])
+const isOpen = ref(false)
+
+const handleIsOpen = () => {
+  isOpen.value = !isOpen.value
+}
+const handleSetValueSelected = (newValue: { label: string; value: string }) => {
```

Comment:
Định nghĩa một cái interface đi em, đặt trong ```interfaces/common.ts```
```suggestion
const handleSetValueSelected = (newValue: SelectOption) => {
```